scripts/dts/extract/globals.py

```diff
@@ -412,91 +437,85 @@ def extract_controller(node_address, prop, prop_values, index,
-        l_cellname = [str(cell_yaml[cell_yaml_names][i]).upper()]
-        if l_cell == l_cellname:
-            label = l_base + l_cell + l_idx
-        else:
-            label = l_base + l_cell + l_cellname + l_idx
-        label_name = l_base + [name] + l_cellname
-        add_compat_alias(node_address, '_'.join(label[1:]), '_'.join(label), prop_alias)
-        prop_def['_'.join(label)] = prop_values.pop(0)
-        if len(name):
-            prop_alias['_'.join(label_name)] = '_'.join(label)
+        try:
+            cell_yaml = get_binding(cell_parent)
+        except:
+            raise Exception(
+                "Could not find yaml description for " +
+                    reduced[cell_parent]['name'])
 
-        # generate defs for node aliases
-        if node_address in aliases:
-            add_prop_aliases(
-                node_address,
-                lambda alias:
-                    '_'.join([convert_string_to_label(alias)] + label[1:]),
-                '_'.join(label),
-                prop_alias)
+        try:
+            name = names.pop(0).upper()
+        except:
+            name = ''
 
-        insert_defs(node_address, prop_def, prop_alias)
+        # Get number of cells per element of current property
+        for k in reduced[cell_parent]['props'].keys():
+            if k[0] == '#' and '-cells' in k:
+                if k in cell_yaml.keys():
+                    cell_yaml_names = k
+                else:
+                    cell_yaml_names = '#cells'
+        try:
+            generation = get_binding(node_address)['properties'][prop
+                    ]['generation']
+        except:
+            generation = ''
 
-    # recurse if we have anything left
-    if not handle_single and len(prop_values):
-        extract_cells(node_address, prop, prop_values, names,
-                      index + 1, def_label, generic)
+        if 'use-prop-name' in generation:
+            l_cell = [convert_string_to_label(str(prop))]
+        else:
+            l_cell = [convert_string_to_label(str(generic))]
 
+        l_base = def_label.split('/')
+        # Check if #define should be indexed (_0, _1, ...)
+        if handle_single or i == 0 and len(p) == 1:
+            # Less than 2 elements in prop_values
+            # Indexing is not needed
+            l_idx = []
+        else:
+            l_idx = [str(i)]
 
+        prop_def = {}
+        prop_alias = {}
 
+        # Generate label for each field of the property element
+        for j in range(num_cells-1):
+            l_cellname = [str(cell_yaml[cell_yaml_names][j]).upper()]
+            if l_cell == l_cellname:
+                label = l_base + l_cell + l_idx
+            else:
+                label = l_base + l_cell + l_cellname + l_idx
+            label_name = l_base + [name] + l_cellname
+            add_compat_alias(node_address, '_'.join(label[1:]), '_'.join(label), prop_alias)
+            prop_def['_'.join(label)] = p[i][j+1]
+            if len(name):
```

Comment:
Can do `if name:` to test if a string/list is empty.

Reply:
Done